src/data/serializer/XYZ.js

```diff
@@ -62,13 +68,36 @@ Ext.define('GeoExt.data.serializer.XYZ', {
             var serialized = {
                 baseURL: source.getUrls()[0],
                 opacity: layer.getOpacity(),
+                imageExtension: this._getImageExtensionFromSource(source)
+                    || 'png',
                 resolutions: tileGrid.getResolutions(),
                 tileSize: ol.size.toSize(tileGrid.getTileSize()),
                 type: "OSM"
             };
             return serialized;
+        },
+
+        /**
+         * Returns the file extension from the url and compares it to whitelist.
+         * Sources with an tileUrlFunction are currently not supported.
+         *
+         * @private
+         * @param {ol.Source} source An ol.Source.XYZ.
+         * @return {String} The fileExtension or false if no one is found.
+         */
+        _getImageExtensionFromSource: function(source){
```

Comment:
>AFAICT we do not use this elsewhere in the library.

I was thinking the same first. But we have several places (partly introduced by me) where we do it. I am indifferent for this PR and think we should remove this in the long run (everywhere?). But this is a discussion we should probably have elsewhere.

Do you agree?